### PR TITLE
[V2] feat: Making scheduler extender node cache capable

### PR DESCRIPTION
**Pull request by landreasyan**
**What type of PR is this?**
Current configuration of scheduler extender was setting `nodeCacheCapable=false` which was resulting in the scheduler passing unnecessary node metadata to the extender. This PR sets `nodeCacheCapable=true` and updates the extender's implementation to not rely on NodeList value passed from the scheduler. 

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/azdiskschedulerextender/main.go [hunk 1]
```diff
@@ -116,13 +116,6 @@ func handleFilterRequest(response http.ResponseWriter, request *http.Request) {
 		return
 	}
 
-	allNodes := args.Nodes.Items
-	if len(allNodes) == 0 {
```
Comment (edreed) [inline pkg/azdiskschedulerextender/main.go:120]: Should we make an equivalent check of `args.NodeNames`?

File: charts/latest-v2/azuredisk-csi-driver/templates/csi-azuredisk-scheduler-extender.yaml [hunk 2]
```diff
@@ -24,7 +24,7 @@ data:
       filterVerb: "filter"
       prioritizeVerb: "prioritize"
       weight: 1
-      nodeCacheCapable: false
```
Comment (edreed) [inline charts/latest-v2/azuredisk-csi-driver/templates/csi-azuredisk-scheduler-extender.yaml:27]: We need to update [deploy/latest-v2/csi-azuredisk-scheduler-extender.yaml](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/main_v2/deploy/latest-v2/csi-azuredisk-scheduler-extender.yaml#L26) as well.

File: pkg/azdiskschedulerextender/main.go (hunk 1 above)
Comment (landreasyan) [reply pkg/azdiskschedulerextender/main.go:120]: I am not sure we need this check in main and respond with BadRequest, we were not doing this for prioritize. I think if nodes are empty, we should just return an empty list?

Comment (edreed) [reply pkg/azdiskschedulerextender/main.go:120]: I'm not sure what the expected behavior is if an empty list is passed by kube-scheduler here. Given that we don't generally fail operations on unexpected input and simply return what was given to us, returning an empty list in this case seems reasonable.

**Review by abhisheksinghbaghel — APPROVED** [review]

**Pull request merged**